docker/Dockerfile

```diff
@@ -12,43 +12,267 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM ubuntu:focal AS base
+# Create basic runtime environment base on Ubuntu 22.04 (jammy)
+# Define build arguments
+# Create and activate runtime virtual environment
+FROM ubuntu:jammy AS base-runtime
+ARG KOKKOS_VERSION=4.1.00
+ARG LIGHTNING_VERSION=0.32.0
+ARG CUDA_VERSION=12.2
+ARG CUDA_ARCH=AMPERE80
+ARG AMD_ARCH=AMD_GFX90A
```

Comment:
I assume these are all user-configurable?

Reply:
Yes. For example,
```
docker build --build-arg="LIGHTNING_VERSION=0.31.0" -f docker/Dockerfile --tag=pennylaneai/pennylane:ligthning-qubit --target wheel-ligthning-qubit .
```
should build v0.31.